addons/l10n_es_website_sale/controllers/main.py

```diff
@@ -0,0 +1,43 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+
+
+class L10nESWebsiteSale(WebsiteSale):
+
+    def _validate_address_values(self, address_values, partner_sudo, address_type, use_delivery_as_billing, *args, **kwargs):
+        invalid_fields, missing_fields, error_messages = super()._validate_address_values(
+            address_values, partner_sudo, address_type, use_delivery_as_billing, *args, **kwargs
+        )
+
+        return invalid_fields, missing_fields, error_messages
+
+    def _get_mandatory_address_fields(self, country_sudo):
+        field_names = super()._get_mandatory_address_fields(country_sudo)
+
+        if self.env.company.country_code == country_sudo.code == 'ES':
```

Comment:
Shouldn't we be using `request` instead of `self.env` ?  Seems to be the pattern used in `website_sale`.

(applies everywhere else)